src/Aspire.Hosting/ApplicationModel/InputReference.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an input reference for a resource with inputs.
+/// </summary>
+/// <param name="owner">The resource with inputs that owns the input.</param>
+/// <param name="input">The <see cref="InputAnnotation"/>.</param>
+public sealed class InputReference(IResource owner, InputAnnotation input) : IManifestExpressionProvider
```

Comment:
Should this just take a `string inputName`, and then have logic to look up the `InputAnnotation` by name from the `Owner` on demand?

Reply:
Yes, model it after EndpointReference.

Reply:
Done.